### PR TITLE
Sw 251 sorting handlers productnumbersearch

### PR DESCRIPTION
**Pull request by zep-com**

### CODE REVIEW

File: FinSearchUnified/Bundle/SearchBundleFindologic/FindologicSortingHandlerInterface.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic;
+
+use Shopware\Bundle\SearchBundle\SortingInterface;
+use Shopware\Bundle\StoreFrontBundle\Struct\ShopContextInterface;
+
+interface FindologicSortingHandlerInterface
```
Comment (ihanli) [inline FinSearchUnified/Bundle/SearchBundleFindologic/FindologicSortingHandlerInterface.php]: Rename this interface to `SortingHandlerInterface`. Its namespace is already mentioning the relation to FINDOLOGIC multiple times.

File: FinSearchUnified/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandler.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Handles the passed sorting object.
+     *
+     * @param SortingInterface $sorting
+     * @param QueryBuilder $query
+     * @param ShopContextInterface $context
+     */
+    public function generateSorting(SortingInterface $sorting, QueryBuilder $query, ShopContextInterface $context)
+    {
+        /** @var PopularitySorting $sorting */
```
Comment (ihanli) [inline FinSearchUnified/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandler.php]: Why is this necessary? Shouldn't the IDE already know about the accessible methods (since it implements the interface)?

Please have a second look and remove the line from each sorting handler if it isn't needed.

File: FinSearchUnified/Tests/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+use Shopware\Components\Test\Plugin\TestCase;
+
+class PopularitySortingHandlerTest extends TestCase
+{
+    /**
+     * @var QueryBuilder
+     */
+    private $querybuilder;
+
+    private $context;
```
Comment (ihanli) [inline FinSearchUnified/Tests/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandlerTest.php]: Add the missing PHPDoc (applies for each test class).

File: FinSearchUnified/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandler.php (hunk 2 above)
Comment (zep-com) [reply FinSearchUnified/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandler.php]: Please check the comment for the same in condition handler PR

Comment (ihanli) [reply FinSearchUnified/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandler.php]: Ah, okay. It would work without it if we used the abstract sorting class instead of the interface. Anyhow, it can stay the way it is.
